code/modules/mob/living/simple_animal/friendly/cat.dm

```diff
@@ -221,22 +223,35 @@
 			else
 				manual_emote(pick("grooms [p_their()] fur.", "twitches [p_their()] whiskers.", "shakes out [p_their()] coat."))
 
-	//MICE!
+	//MICE! RATS! OH MY!
 	if((src.loc) && isturf(src.loc))
 		if(!stat && !resting && !buckled)
-			for(var/mob/living/simple_animal/mouse/M in view(1,src))
-				if(istype(M, /mob/living/simple_animal/mouse/brown/tom) && inept_hunter)
-					if(COOLDOWN_FINISHED(src, emote_cooldown))
-						visible_message(span_warning("[src] chases [M] around, to no avail!"))
-						step(M, pick(GLOB.cardinals))
-						COOLDOWN_START(src, emote_cooldown, 1 MINUTES)
-					break
+			//Targeting anything in the rat faction nearby
+			for(var/mob/living/M in view(1,src))
 				if(!M.stat && Adjacent(M))
-					manual_emote("splats \the [M]!")
-					M.splat()
-					movement_target = null
-					stop_automated_movement = 0
-					break
+					if (faction_check(M.faction.Copy(), list("rat")))
+						//Jerry can never catch Tom snowflaking
+						if(istype(M, /mob/living/simple_animal/mouse/brown/tom) && inept_hunter)
+							if(COOLDOWN_FINISHED(src, emote_cooldown))
+								visible_message(span_warning("[src] chases [M] around, to no avail!"))
+								step(M, pick(GLOB.cardinals))
+								COOLDOWN_START(src, emote_cooldown, 1 MINUTES)
+							break
+						//Mouse splatting
+						if(istype(M, /mob/living/simple_animal/mouse))
+							manual_emote("splats \the [M]!")
+							var/mob/living/simple_animal/mouse/snack = M
+							snack.splat()
+							movement_target = null
+							stop_automated_movement = 0
+							break
+						//Rat scratching, or anything else that could be in the rat faction
+						//Copied from hostile simple animal code
+						if(SEND_SIGNAL(src, COMSIG_HOSTILE_PRE_ATTACKINGTARGET, M) & COMPONENT_HOSTILE_NO_ATTACK)
+							return FALSE //but more importantly return before attack_animal called
+						var/result = M.attack_animal(src)
+						SEND_SIGNAL(src, COMSIG_HOSTILE_POST_ATTACKINGTARGET, M, result)
```

Comment:
Signals should only be fired from one place.

This should instead be a shared proc that hostile code uses.

Reply:
Looking at this again, I didn't need to have the signals there (and copy that code in the first place) as they're mainly for hostile animal AI behavior, getting rid of everything but attack_animal still works and doesn't appear to break anything when testing.

I was also debating refactoring cats to hostile/retaliate with this PR, but I don't think that's necessary right?